helm/cluster-api-core/templates/controller-deployment.yaml

```diff
@@ -34,7 +34,7 @@ spec:
       - args:
         - --metrics-addr=127.0.0.1:{{ .Values.ports.metrics }}
         - --feature-gates=MachinePool={{ .Values.featuregates.machinepool }},ClusterResourceSet={{ .Values.featuregates.clusterresourceset }}
-        - --watch-filter=0.0.1
+        - --watch-filter={{ .Chart.Version }}
```

Comment:
Does this actually work? I remember the Chart stuff to be somehow special

Reply:
I haven't tested it. Maybe I'll just try and see how it looks?